src/Command/SubProfileCommand.php

```diff
@@ -0,0 +1,351 @@
+    $dependencies = $this->validator->validateModuleDependencies($input->getOption('dependencies'));
+    if ($dependencies) {
+      // @todo ProfileCommand::checkDependencies is private so we aren't
+      // actually checking to see if they are present.
+      $dependencies = $dependencies['success'];
+    }
+    $this->buildExcludedDependenciesList($input->getOption('excluded_dependencies'));
+    $this->buildExcludedDependenciesList($input->getOption('excluded_subcomponents'));
+    $this->generator->generate(
+      $profile,
+      $machine_name,
+      $profile_path,
+      $description,
+      $dependencies,
+      $this->excludedDependencies
+    );
+  }
+
+  /**
+   * @param array $excluded_dependencies
+   *   Dependencies to exclude.
+   *
+   * Adds the provided component(s) to the excluded list and, if the provided
+   * component is a top-level component, all of its subcomponents too.
+   */
+  protected function buildExcludedDependenciesList($excluded_dependencies) {
+    $excluded_dependencies_list = [];
+    foreach ($excluded_dependencies as $excluded_dependency) {
+      if (array_key_exists($excluded_dependency, $this->getLightningComponents())) {
+        $excluded_dependencies_list[] = $excluded_dependency;
+        if (in_array($excluded_dependency, $this->getTopLevelComponents())) {
+          // If its a top-level-component, add its subcomponents too.
+          $subcomponents = $this->getSubComponents(trim($excluded_dependency));
+          foreach ($subcomponents as $subcomponent) {
+            $excluded_dependencies_list[] = $subcomponent;
+          }
+        }
+      }
+    }
+    $this->excludedDependencies = array_merge($this->excludedDependencies, $excluded_dependencies_list);
+  }
+
+  public static function getLightningComponents() {
+    $appRoot = \Drupal::root();
+    $extensions = new ExtensionDiscovery($appRoot);
+    $extensions = $extensions->scan('module');
+
+    $lightning_extensions = self::array_filter_key($extensions, function($key) {
+      return strpos($key, 'lightning_') === 0;
+    });
+
+    $lightning_components = [];
+    foreach($lightning_extensions as $machine_name => $lightning_extension) {
+      $InfoParser = new InfoParser();
+      $info = $InfoParser->parse($lightning_extension->getPathname());
+      $lightning_components[$machine_name] = [
+        'name' => $info['name'],
+      ];
+      if (isset($info['components'])) {
+        $lightning_components[$machine_name]['subcomponents'] =  $info['components'];
+      }
+    }
+
+    return $lightning_components;
+  }
+
+  /**
+   * @param $input
+   * @param $callback
+   * @return array|null
+   *
+   * Filter an array by keys.
+   */
+  private static function array_filter_key(array $input, $callback) {
+    $keys = array_keys($input);
+    $filteredKeys = array_filter($keys, $callback);
+    if (empty($filteredKeys)) {
+      return [];
+    }
+    $input = array_intersect_key($input, array_flip($filteredKeys));
```

Comment:
Nit: Can we return directly from array_intersect_key() rather than overwrite $input, for clarity?